pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,48 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					fldPath.Child("updateStrategy").Child("partition"),
+					spec.UpdateStrategy.Partition.Ordinal,
+					fmt.Sprintf("only allowed for updateStrategy '%s'", apps.PartitionStatefulSetStrategyType)))
+		}
+	case apps.PartitionStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition == nil {
```

Comment:
I expect the approach of using a different strategy for partitioning to be more confusing and painful as opposed to this being an option for rollingUpdate. If I want to build an operator on top of StatefulSets, I am going to lose a lot of UX with the current approach. Assuming we add Conditions in the future and surface one once an update is complete, it is going to be much easier with a single strategy. I am thinking of post hooks. Now, you can't have a post-hook and re-use the StatefulSet controller to set a complete Condition just by switching Partition from 0 (which means all Pods have been updated) to nil. You have to make the operator set the Condition.

cc: tnozicka